lms/djangoapps/commerce/api/v1/models.py

```diff
@@ -88,8 +88,7 @@ def update(self, attrs):
             merged_mode.min_price = posted_mode.min_price
             merged_mode.currency = posted_mode.currency
             merged_mode.sku = posted_mode.sku
-            if posted_mode.expiration_datetime is not None:
```

Comment:
Both these checks were originally not present, and were introduced in https://github.com/edx/edx-platform/pull/10631/files#diff-d5cfd483f2422f69931e73edec3cdfa6. I think this was some confusion around the new deadline defaults which were part of the first round of self-paced work.